attentive-android-sdk/src/main/java/com/attentive/androidsdk/AttentiveApi.java

```diff
@@ -0,0 +1,187 @@
+
+    public void sendUserIdentifiersCollectedEvent(String domain, UserIdentifiers userIdentifiers, AttentiveApiCallback callback) {
+        // first get the geo-adjusted domain, and then call the events endpoint
+        final String url = String.format(ATTENTIVE_DTAG_URL, domain);
+        Request request = new Request.Builder().url(url).build();
+        httpClient.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                callback.onFailure("Getting geo-adjusted domain failed: " + e.getMessage());
+            }
+
+            @Override
+            public void onResponse(@NonNull Call call, @NonNull Response response) throws IOException {
+                // Check explicitly for 200 (instead of response.isSuccessful()) because the response only has the tag in the body when its code is 200
+                if (response.code() != 200) {
+                    callback.onFailure(String.format("Getting geo-adjusted domain returned invalid code: '%d', message: '%s'", response.code(), response.message()));
+                    return;
+                }
+
+                ResponseBody body = response.body();
+
+                if (body == null) {
+                    callback.onFailure("Getting geo-adjusted domain returned no body");
+                    return;
+                }
+
+                String fullTag = response.body().string();
+                String geoAdjustedDomain = parseAttentiveDomainFromTag(fullTag);
+
+                if (geoAdjustedDomain == null) {
+                    callback.onFailure("Could not parse the domain from the full tag");
+                    return;
+                }
+
+                internalSendUserIdentifiersCollectedEventAsync(geoAdjustedDomain, userIdentifiers, callback);
+            }
+        });
+    }
+
+    private void internalSendUserIdentifiersCollectedEventAsync(String geoAdjustedDomain, UserIdentifiers userIdentifiers, AttentiveApiCallback callback) {
+        String externalVendorIdsJson = null;
+        try {
+            List<ExternalVendorId> externalVendorIds = buildExternalVendorIds(userIdentifiers);
+            externalVendorIdsJson = objectMapper.writeValueAsString(externalVendorIds);
+        } catch (JsonProcessingException e) {
+            callback.onFailure(String.format("Could not serialize the UserIdentifiers. Message: '%s'", e.getMessage()));
+            return;
+        }
+
+        String metadataJson = null;
+        try {
+            Metadata metadata = buildMetadata(userIdentifiers);
+            metadataJson = objectMapper.writeValueAsString(metadata);
+        } catch (JsonProcessingException e) {
+            callback.onFailure(String.format("Could not serialize metadata. Message: '%s'", e.getMessage()));
+            return;
+        }
+
+        HttpUrl url = getHttpUrlEventsEndpointBuilder()
+                .addQueryParameter("v", "4.16.10_f0149441bf")
```

Comment:
i think we may just be able to use any string for this param - we can test by sending a test event & seeing if it makes it to snowflake